src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -70,6 +70,18 @@ object TypeCast {
     }
   }
 
+  private def parseXmlBoolean(s: String): Boolean = {
+    if (s != null) s.toLowerCase match {
+      case "true" => true
+      case "false" => false
+      case "1" => true
+      case "0" => false
+      case _ => throw new IllegalArgumentException("For input string: \"" + s + "\"")
+    }
+    else {
+      throw new IllegalArgumentException("For input string: \"null\"")
```

Comment:
Skip this clause and just `s.toLowerCase match`. It will throw an NPE as before.